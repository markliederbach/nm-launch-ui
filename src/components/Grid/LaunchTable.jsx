import React, { Component } from "react";
import { observer } from "mobx-react";
import {
  Table,
  TableHeader,
  TableRow,
  TableHeaderCell,
  TableBody,
  TableCell
} from "semantic-ui-react";

@observer
export class LaunchTable extends Component {
  render() {
    let resultsList = this.props.results.launches.map((launch, index) => (
      <TableRow key={index}>
        <TableCell>{launch.est_timestamp}</TableCell>
        <TableCell>{launch.location.name}</TableCell>
        <TableCell>{launch.rocket.name}</TableCell>
      </TableRow>
    ));

    return (
      <Table>
        <TableHeader>
          <TableRow>
            <TableHeaderCell>Date</TableHeaderCell>
            <TableHeaderCell>Location</TableHeaderCell>
            <TableHeaderCell>Rocket</TableHeaderCell>
          </TableRow>
        </TableHeader>
        <TableBody>{resultsList}</TableBody>
      </Table>
    );
  }
}
