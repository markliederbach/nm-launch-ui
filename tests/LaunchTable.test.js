import React from "react";
import { shallow, configure } from "enzyme";
import renderer from "react-test-renderer";
import { LaunchTable } from "../src/components/Grid/LaunchTable";
import moment from "moment-timezone";
import Adapter from "enzyme-adapter-react-16";

configure({ adapter: new Adapter() });

describe("LaunchTable", function() {
  let defaultProps;
  beforeAll(() => {
    moment.tz.setDefault("CST");
    defaultProps = {
      results: {
        launches: [
          {
            est_timestamp: moment("2018-09-27T09:34:00").format(
              "YYYY-MM-DDTHH:mm:ss"
            ),
            name: "Some launch name",
            location: {
              name: "location name",
              map_url: "http://example.com"
            },
            missions: [
              {
                agencies: [
                  {
                    name: "NASA"
                  },
                  {
                    name: "ESA"
                  }
                ]
              }
            ],
            rocket: {
              name: "Falcon 9",
              info_url: "http://example2.com"
            }
          }
        ]
      }
    };
  });

  it("renders props correctly", () => {
    const tree = renderer.create(<LaunchTable {...defaultProps} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
