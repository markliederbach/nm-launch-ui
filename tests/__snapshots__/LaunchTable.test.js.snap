// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LaunchTable renders props correctly 1`] = `
<table
  className="ui table"
>
  <thead
    className=""
  >
    <tr
      className=""
    >
      <th
        className=""
      >
        Time (UTC)
      </th>
      <th
        className=""
      >
        Name
      </th>
      <th
        className=""
      >
        Agency
      </th>
      <th
        className=""
      >
        Location
      </th>
      <th
        className=""
      >
        Rocket
      </th>
    </tr>
  </thead>
  <tbody
    className=""
  >
    <tr
      className=""
    >
      <td
        className=""
      >
        <time
          dateTime={"2018-09-27T09:34.000Z"}
        >
          Thu Sep 27 2018 9:34AM
        </time>
      </td>
      <td
        className=""
      >
        Some launch name
      </td>
      <td
        className=""
      >
        NASA
      </td>
      <td
        className=""
      >
        <a
          href="http://example.com"
          rel="noopener"
          target="_blank"
        >
          location name
        </a>
      </td>
      <td
        className=""
      >
        <a
          href="http://example2.com"
          rel="noopener"
          target="_blank"
        >
          Falcon 9
        </a>
      </td>
    </tr>
  </tbody>
</table>
`;
